src/System.Management/src/System/Management/ManagementScope.cs

```diff
@@ -304,332 +303,145 @@ static WmiNetUtilsHelper()
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "ExecQueryWmi");
-                if( procAddr != IntPtr.Zero)
-                {
-                    ExecQueryWmi_f  =(ExecQueryWmi) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(ExecQueryWmi));
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "ExecNotificationQueryWmi");
-                if( procAddr != IntPtr.Zero)
-                {
-                    ExecNotificationQueryWmi_f  =(ExecNotificationQueryWmi) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(ExecNotificationQueryWmi));
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "PutInstanceWmi");
-                if( procAddr != IntPtr.Zero)
-                {
-                    PutInstanceWmi_f  =(PutInstanceWmi) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(PutInstanceWmi));
-                } 
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "PutClassWmi");
-                if( procAddr != IntPtr.Zero)
-                {
-                    PutClassWmi_f  =(PutClassWmi) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(PutClassWmi));
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "CloneEnumWbemClassObject");
-                if( procAddr != IntPtr.Zero)
-                {
-                    CloneEnumWbemClassObject_f  =(CloneEnumWbemClassObject) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(CloneEnumWbemClassObject));
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "ConnectServerWmi");
-                if( procAddr != IntPtr.Zero)
-                {
-                    ConnectServerWmi_f  =(ConnectServerWmi) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(ConnectServerWmi));
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "GetErrorInfo");
-                if( procAddr != IntPtr.Zero)
-                {
-                    GetErrorInfo_f  =(GetErrorInfo) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(GetErrorInfo));
-                }
-                procAddr = Interop.Kernel32.GetProcAddress(loadLibrary, "Initialize");
-                if( procAddr != IntPtr.Zero)
-                {
-                    Initialize_f  =(Initialize) Marshal.GetDelegateForFunctionPointer(procAddr, typeof(Initialize));
-                }
+                // This is unlikely, so having the TypeInitializationException wrapping it is fine.
+                throw new Win32Exception(Marshal.GetLastWin32Error(), string.Format(SR.LoadLibraryFailed, wminet_utilsPath));
+            }
 
-        Initialize_f(CompatSwitches.AllowIManagementObjectQI);
+            if (LoadDelegate(ref ResetSecurity_f, Interop.Kernel32.GetProcAddress(loadLibrary, "ResetSecurity")) &&
```

Comment:
Move the `Interop.Kernel32.GetProcAddress` call into the LoadDelegate method to make this code more compact.